static/js/blueslip_stacktrace.ts

```diff
@@ -0,0 +1,122 @@
+import $ from 'jquery';
+import _ from 'underscore';
+import { SourceMapConsumer, IndexedSourceMapConsumer } from 'source-map';
+import StackTraceGPS, { StackFrame } from 'stacktrace-gps';
+import ErrorStackParser from 'error-stack-parser';
+const render_blueslip_stacktrace = require('../templates/blueslip_stacktrace.hbs');
+const source_map_mappings = require('source-map/lib/mappings.wasm');
+
+SourceMapConsumer.initialize({
+    "lib/mappings.wasm": source_map_mappings,
+});
+
+type CleanStackFrame = {
+    full_path: string;
+    show_path: string;
+    function_name: string;
+    function_scope: string;
+    line_number: number;
+};
+
+export function clean_stackframe_data(data: StackFrame): CleanStackFrame {
+    // If the file is local, just show the filename.
+    // Otherwise, show the full path starting from node_modules.
+    const idx = data.fileName.indexOf('node_modules');
+    let full_path;
+    let show_path;
+
+    if (idx !== -1) {
+        full_path = show_path = data.fileName.slice(idx);
+    } else {
+        full_path = data.fileName.split('/').slice(-3).join('/');
+        show_path = full_path.slice(full_path.lastIndexOf('/') + 1);
+    }
+
+    // Don't show the full scope in function name
+    let full_fn_name;
+    try {
+        full_fn_name = /(?:[a-z_]+\.[jt]s(?:\?\w+)?\.)(.+)/.exec(data.functionName)[1];
+    } catch {
+        full_fn_name = data.functionName;
+    }
+    const fn_name = full_fn_name.split('.');
+
+    return {
+        full_path: full_path,
+        show_path: show_path,
+        line_number: data.lineNumber,
+        function_name: fn_name.pop(),
+        function_scope: fn_name.join('.'),
+    };
+}
+
+// The stacktrace-gps library relies on the mozilla/source-map WASM library,
+// we assume the developer's browser is recent enough to support it.
+const stack_trace_gps = new StackTraceGPS();
```

Comment:
Now I’m curious…how does this library avoid doing the `.wasm` URL dance we just had to do?

Reply:
They are using an old version of source-map which doesn't use the wasm mappings.